eng/native/configureplatform.cmake

```diff
@@ -492,10 +492,3 @@ if(NOT CLR_CMAKE_TARGET_BROWSER AND NOT CLR_CMAKE_TARGET_WASI)
 
     set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 endif()
-
-string(TOLOWER "${CMAKE_BUILD_TYPE}" LOWERCASE_CMAKE_BUILD_TYPE)
-if(LOWERCASE_CMAKE_BUILD_TYPE STREQUAL debug)
-    # Clear _FORTIFY_SOURCE=2, if set
```

Comment:
Why was this needed and why is it not necessary anymore?

Reply:
This was done for Clear Linux OS, which sets `_FORTIFY_SOURCE=2` globally.

From the [article](https://developers.redhat.com/articles/2022/06/02/use-compiler-flags-stack-protection-gcc-and-clang#)
> GCC and Clang select fortified sources using the preprocessor flag `-D_FORTIFY_SOURCE=<n>` with an optimization level greater than `-O0`. A higher <n> corresponds to greater protection. However, there may be a tradeoff in performance, and some programs that conform to the standard may fail due to stricter security checks. GCC ([version 12](https://developers.redhat.com/articles/2022/04/25/new-c-features-gcc-12) and later) and Clang (version 9.0 and later) support `<n>` up to 3.

The debug build (`-O0`) on Clear Linux was failing: https://github.com/dotnet/runtime/issues/2188.

This PR is moving this to configurecompiler.cmake.